src/rez/package_resources_.py

```diff
@@ -429,9 +437,26 @@ def _uri(self):
     def _subpath(self):
         if self.index is None:
             return None
+
+        if self.parent.hashed_variants:
+            h = sha1(str(map(str, self.variant_requires)))
```

Comment:
One issue to think about, especially if you're planning on eventually supporting python 3, is that with the str/unicode/bytes change in python 3, the arguments to the hashlib hashes are a bit different, and they no longer accept unicode strings, only encoded byte strings.  And ideally, you'd want the hash to be the same whether it was run from a python 2 or python 3 rez.

The following should produce the same hash in py2/3:

```
import six
variant = ['one', 'two']
h = sha256()
h.update(u','.join(map(six.text_type, variant)).encode('utf8'))
h.digest()
```